### ISSUE TITLE
Channel Permissions

### ISSUE THREAD

**Issue by Ballaual** [HunteRoi/discord-temp-channels#10]
Hey Mate,

you did very well with this implementation!

I am currently working a possible solution to get this working with SlashCommand (as far as i got it works very well).
But now i am struggle to keep the parent permissions. As i see you create a blank channel, copy permissions from the category and set the user as owner to edit it.

Do you think you might be able to keep the permissions from the parent channel instead of the category?
I think this might be more useful than using the category ones.
Not sure if this might be same as that issue: https://github.com/HunteRoi/discord-temp-channels/issues/8

Best regards! :)
Alex

**Comment by Ballaual** [comment]
That's my implementation using slashcommands, just for your information
Actually missing a command to delete the trigger again and some logic to handle with created channels if bot crashes.
-> Delete when empty

Also i use a scuffed method to clone the bitrate and max users from the parent channel.

[loadTempChannels.txt](https://github.com/HunteRoi/discord-temp-channels/files/11201320/loadTempChannels.txt)
[registerChannel.txt](https://github.com/HunteRoi/discord-temp-channels/files/11201252/registerChannel.txt)

**Comment by Ballaual** [comment]
@HunteRoi just in case i dont know if u would read this without ping :)

**Comment by HunteRoi** [comment]
Hey! 

Thanks for reaching out! If I understand your concern, this [piece of code](https://github.com/HunteRoi/discord-temp-channels/blob/master/src/TempChannelsManager.ts#L168-L174) uses the category of the parent to set the permissions for the channel, but you'd rather use the parent's permissions directly, right?

I think I can add a parameter to either choose the permissions from the category or from the parent. Would that be good?

Thank you again for the interest in this package, I really appreciate to see that my work is used by others 😄 


EDIT: or maybe you'd rather see [this line](https://github.com/HunteRoi/discord-temp-channels/blob/6e23b7b82120058cd83a3692200a700105585fbe/src/TempChannelsManager.ts#L182) updated to use the parent's permissions instead of the category? That would be a breaking change for other users anyway, so I will have to introduce a new option anyways ^^

**Comment by Ballaual** [comment]
Hi, thank you for your fast reply :) and sorry for my bad english :D
I guess you understood what my problem is ^^

To get this working as intended i changed the logic to this, might be causing some issues (will point that out later)

```
      const voiceChannel = await channel.guild.channels.create({
            name,
            parent: categoryChannel?.id ?? null,
            bitrate: parentChannel.bitrate,
            userLimit: parentChannel.userLimit,
            type: ChannelType.GuildVoice,
            permissionOverwrites: parentChannel ? [...parentChannel.permissionOverwrites.cache.values()] : []
        });
```

But i will provide you bit more information in the screenshot to clarify my very specific request :)

The main problem for me was/is that this implementation is fully hardcoded and not dynamic.
Thats totally fine if you use the bot only on one server but not when you want to provide a public bot that might be used on mutliple server.

![Unbenannt](https://user-images.githubusercontent.com/38478976/231461204-b7b22b4d-5d26-4715-98bb-5efd42b617be.png)

For using this in a dynamic way i tryed using slashcommand register and unregister channels using your package and store them into a sqlite database. I got this working so far but seems to be a little buggy sometimes.. When joining the parent channel and creating the child channel it sometimes double creates and deletes the other one immediately.

When restarting the bot the parent channels still work but there is still a problem that created childs wont be deleted.
As i am relativly new to js and developemtn i am not sure if that could by editing the package or needs to be coded somewhere in the codebase?!

If you want to take a look yourself you can find the Repo DiCoBo on my Github.
Any help / advise appreciated :)

**Comment by HunteRoi** [comment]
New version available with the parameter introduced by #11 🚀😃
